From: Roger Shimizu <rosh@debian.org>
Date: Sun, 26 Apr 2020 18:25:52 +0900
Subject: Fix use case that enforce install of EN version on non-EN user
 session

build_paths() is called before load_settings(), so it won't recognize
"force_en-US" setting.

Here we add build_paths() again after load_settings() to do the trick.

Closes: #842021
---
 torbrowser_launcher/common.py | 15 ++++++++-------
 1 file changed, 8 insertions(+), 7 deletions(-)

diff --git a/torbrowser_launcher/common.py b/torbrowser_launcher/common.py
index 6533723..1679a96 100644
--- a/torbrowser_launcher/common.py
+++ b/torbrowser_launcher/common.py
@@ -62,6 +62,7 @@ class Common(object):
             self.mkdir(self.paths['dirs'][d])
         self.load_mirrors()
         self.load_settings()
+        self.build_paths()
         self.mkdir(self.paths['download_dir'])
         self.mkdir(self.paths['tbb']['dir'])
         self.init_gnupg()
@@ -107,6 +108,10 @@ class Common(object):
         tbb_local = '{0}/.local/share/torbrowser'.format(homedir)
         old_tbb_data = '{0}/.torbrowser'.format(homedir)
 
+        if hasattr(self, 'settings') and self.settings['force_en-US']:
+            language = 'en-US'
+        else:
+            language = self.language
         if tbb_version:
             # tarball filename
             if self.architecture == 'x86_64':
@@ -114,10 +119,6 @@ class Common(object):
             else:
                 arch = 'linux32'
 
-            if hasattr(self, 'settings') and self.settings['force_en-US']:
-                language = 'en-US'
-            else:
-                language = self.language
             tarball_filename = 'tor-browser-' + arch + '-' + tbb_version + '_' + language + '.tar.xz'
 
             # tarball
@@ -153,11 +154,11 @@ class Common(object):
                 'version_check_file': tbb_cache + '/download/release.xml',
                 'tbb': {
                     'changelog': tbb_local + '/tbb/' + self.architecture + '/tor-browser_' +
-                                 self.language + '/Browser/TorBrowser/Docs/ChangeLog.txt',
+                                 language + '/Browser/TorBrowser/Docs/ChangeLog.txt',
                     'dir': tbb_local + '/tbb/' + self.architecture,
-                    'dir_tbb': tbb_local + '/tbb/' + self.architecture + '/tor-browser_' + self.language,
+                    'dir_tbb': tbb_local + '/tbb/' + self.architecture + '/tor-browser_' + language,
                     'start': tbb_local + '/tbb/' + self.architecture + '/tor-browser_' +
-                             self.language + '/start-tor-browser.desktop'
+                             language + '/start-tor-browser.desktop'
                 },
             }
 
